src/csharp/Grpc.Core/AsyncClientStreamingCall.cs

```diff
@@ -36,7 +36,21 @@ public sealed class AsyncClientStreamingCall<TRequest, TResponse> : IDisposable
         readonly Func<Metadata> getTrailersFunc;
         readonly Action disposeAction;
 
-        internal AsyncClientStreamingCall(IClientStreamWriter<TRequest> requestStream, Task<TResponse> responseAsync, Task<Metadata> responseHeadersAsync, Func<Status> getStatusFunc, Func<Metadata> getTrailersFunc, Action disposeAction)
+        /// <summary>
+        /// Creates a new AsyncClientStreamingCall object with the specified properties.
+        /// </summary>
+        /// <param name="requestStream">Stream of request values.</param>
+        /// <param name="responseAsync">The response of the asynchronous call.</param>
+        /// <param name="responseHeadersAsync">Response headers of the asynchronous call.</param>
+        /// <param name="getStatusFunc">Delegate returning the status of the call.</param>
+        /// <param name="getTrailersFunc">Delegate returning the trailing metadata of the call.</param>
+        /// <param name="disposeAction">Delegate to invoke when Dispose is called on the call object.</param>
+        public AsyncClientStreamingCall(IClientStreamWriter<TRequest> requestStream,
+                                        Task<TResponse> responseAsync,
+                                        Task<Metadata> responseHeadersAsync,
```

Comment:
All 4 call types  share these 4 arguments:
`Task<Metadata> responseHeadersAsync, Func<Status> getStatusFunc, Func<Metadata> getTrailersFunc, Action disposeAction`.
Wouldn't it be useful to wrap them in a single class (I am actually undecided what's better here)?

Reply:
The main advantage of having a class is future proofing the signature, so if we can add stuff in it in the future (which I think is a worthy goal), but if we are confident about the signature, I don't think wrapping in a class is more user-friendly here. What do you think?